src/sentry/deletions/defaults/detector.py

```diff
@@ -0,0 +1,19 @@
+from sentry.deletions.base import BaseRelation, ModelDeletionTask, ModelRelation
+from sentry.workflow_engine.models.detector import Detector
+
+
+class DetectorDeletionTask(ModelDeletionTask[Detector]):
+    def get_child_relations(self, instance: Detector) -> list[BaseRelation]:
+        from sentry.workflow_engine.models import DataConditionGroup, DataSource
+
+        # XXX: this assumes a data source is connected to a single detector. it's not possible in the UI
+        # to do anything else today, but if this changes we'll need to add custom conditional deletion logic
+
+        model_relations: list[BaseRelation] = [ModelRelation(DataSource, {"detector": instance.id})]
+
+        if instance.workflow_condition_group:
+            model_relations.append(
+                ModelRelation(DataConditionGroup, {"id": instance.workflow_condition_group.id})
+            )
```

Comment:
Should we perform a failsafe here to validate that there aren't any other rows related to the data source? This seems like a footgun otherwise, I doubt people will remember to check the deletions

Reply:
I found while working on this that I can't perform db queries in the deletion task otherwise some tests in `tests/sentry/db/test_silo_models.py`. Is there another place that would make sense for me to check that? The only thing I can think of is a post delete signal maybe.

Reply:
That's a bit confusing... we're doing queries here to fetch the rows that should be deleted, what failures were you seeing?

Reply:
`test_cross_silo_deletions` fails with a `RuntimeError: Database access not allowed, use the "django_db" mark, or the "db" or "transactional_db" fixtures to enable it.` if I add queries. 

I'm having a hard time understanding exactly what's going on - for example if I add `data_sources = DataSource.objects.filter(detector=instance.id)` that's fine, but the second I try to print `data_sources[0]` or in any way access the data from it (like checking if there are other detectors using the data source- `Detector.objects.filter(data_sources__in=[data_source.id for data_source in data_sources]).count()` it fails with the db access not allowed error.

Reply:
I think you just need to add `@django_db_all` to that test, it just isn't expecting to make db calls

Reply:
Oh I assumed it was set up like that on purpose, but if you think it's ok I'll add that 🤷

Reply:
I think they just didn't need the db so they didn't bother adding the tag. I don't see any reason why it can't be there though